plugins/modules/mso_schema_site_service_graph.py

```diff
@@ -225,29 +280,77 @@ def main():
             )
 
         if devices is not None:
-            service_node_type_names_from_template = [type.get("name") for type in service_node_types_from_template]
+            query_device_data = mso.lookup_service_node_device(site_id, tenant, device_name=None, service_node_type=None)
             for index, device in enumerate(devices):
-                template_node_type = service_node_type_names_from_template[index]
-                apic_type = "OTHERS"
-                if template_node_type == "firewall":
-                    apic_type = "FW"
-                elif template_node_type == "load-balancer":
-                    apic_type = "ADC"
-                query_device_data = mso.lookup_service_node_device(site_id, tenant, device.get("name"), apic_type)
-                devices_payload.append(
-                    dict(
-                        device=dict(
-                            dn=query_device_data.get("dn"),
-                            funcTyp=query_device_data.get("funcType"),
-                        ),
-                        serviceNodeRef=dict(
-                            serviceNodeName=template_node_type,
-                            serviceGraphName=service_graph,
-                            templateName=template,
-                            schemaId=schema_id,
-                        ),
-                    ),
-                )
+                device_payload = dict()
+                if query_device_data:
+                    for device_data in query_device_data:
+                        if device.get("device_name") == device_data.get("dn").split("/")[-1].split("-")[-1]:
```

Comment:
Can `device_data.get("dn").split("/")[-1].split("-")[-1]` ever lead to an error?

Reply:
The DN value points to the Cloud Site L4-L7 device, so it should follow the same pattern. Also, the DN format is common for the current L4-L7 devices.

Reply:
ok